### ISSUE TITLE
Closing multiple unsaved files creates multiple unusable "Do you want to save" dialogues

### ISSUE THREAD

**Issue by MaPePeR** [atom/atom#17459]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
    * Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
    * Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
    * Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description

When closing multiple files with unsaved changes using the "Close all" or similar functionality creates multiple "Do you want to save this file before closing" dialogues. But only one of these dialogues can be used, because the other dialogues are opened "on top" of each other, so every other dialog is blocked until the they are resolved in order.

### Steps to Reproduce

1. Open multiple files and make changes to them without saving
   * <kbd>Ctrl</kbd> + <kbd>n</kbd>, Random letters, <kbd>Ctrl</kbd> + <kbd>n</kbd>, Random letters, <kbd>Ctrl</kbd> + <kbd>n</kbd>, Random letters, ...
2. Right click the tab-bar and select "Close all"

**Expected behavior:** Multiple dialogues are opened, but the order in which i resolve them does not matter. (The main Atom window stays blocked until all dialogues are resolved)

**Actual behavior:** Multiple dialogues are opened, but only one of them reacts to clicks.

**Reproduces how often:** 100%

### Versions

```
$ ./atom --version
Atom    : 1.27.2
Electron: 1.7.15
Chrome  : 58.0.3029.110
Node    : 7.9.0
```

```
$ uname -r
4.16.12-1-ARCH
```

Same issue as #17429, but this time with an official atom release... the issue was not fixed by using older versions of Electron, Chrome and Node, so i created [a new issue as requested by @rsese](https://github.com/atom/atom/issues/17429#issuecomment-393263283)

**Comment by rsese** [comment]
> Same issue as #17429, but this time with an official atom release

Thanks @MaPePeR - I might have missed that you were using Arch in your original issue, how did you install Atom?

**Comment by rsese** [comment]
I don't have Arch to test with but I tried in Ubuntu 16.04 with Atom 1.27.1 and see the same behavior - multiple dialogs are opened and you can click on any of the dialogs themselves, but it looks like you can only click the `Save`, `Cancel`, or `Don't Save` buttons in the dialogs in order of the tabs from right to left.

On Windows 10 and macOS 10.12.6, the behavior is different - after selecting "Close All Tabs" to close multiple tabs, a single dialog pops up for the rightmost tab and after clicking one of the buttons to save, don't save, or cancel, no other dialogs pop up for the rest of the tabs.

**Comment by MaPePeR** [comment]
So on windows and macOS the issue is even worse, because unsaved work is just discarded without even asking? :flushed:

**Comment by rsese** [comment]
No sorry my comment wasn't clearer - I just double checked again and see different behavior on Windows (not sure if I was testing an older version at the time 🤔) but to clarify using the issue's repro steps:

* On macOS 10.12.6 with Atom 1.29.0-dev-0422e0d61: a single dialog pops up for a single file (I think the leftmost tab?) and after you choose save, don't save, or cancel, no more dialogs popup and the other untitled files are left alone.
* On Windows 10 with Atom 1.27.1: a single dialog pops up for a single file and after you choose save, don't save, or cancel, another dialog pops up for the next file (and this continues for each untitled file).

So no unsaved work is discarded without asking.

**Comment by sadick254** [comment]
Thanks for reporting this!

This has already been fixed in the latest version of atom `v1.51.0`. Please consider upgrading to the latest atom.

**Comment by MaPePeR** [comment]
@sadick254 I just downloaded the standalone atom-amd64.tar.gz from the 1.52.0-beta0 release and the issue is not yet fixed in that release on Linux.

Closing multiple tabs with unsaved changes still opens multiple dialog windows at once to confirm closing the files and it's still the case, that only one of these dialog windows is responsive.
